### PR TITLE
updated action page

### LINKED ISSUE #563: update action page
https://app.zeplin.io/project/5c81bede826def625074c12c/screen/5cd79d99188bff348afc51d8

Screen Shot 2019-05-21 at 6 25 41 PM

### PR DESCRIPTION
**Pull request by gyanpgupta**
kind ui-change

Fixes #563

### CODE REVIEW

**Comment by nmhung1210** [comment]
> kind ui-change
> 
> Fixes #563

Please help attach the screenshot of the result for all `ui-change` requests. It's helpful for you to verify the result and reviewer to address the change quickly.

**Comment by gyanpgupta** [comment]
IoTeX Action b181433ba10ee8ad1a0db2977c701e3744a6f07585f4c211e1ac38b742a609d7

File: src/shared/action/action-detail.tsx [hunk 1]
```diff
@@ -1,6 +1,5 @@
+import { Card, Col, Icon, Row } from "antd";
```
Comment (nmhung1210) [inline src/shared/action/action-detail.tsx]: Please import separately package needed from `antd/lib/...` to optimize the bundle size

File: src/shared/action/action-detail.tsx [hunk 2]
```diff
@@ -145,13 +151,57 @@ class ActionDetailsInner extends PureComponent<Props> {
...
   }
 
+  public renderActionData(): JSX.Element {
+    const { dataSource } = this.state;
+    return (
+      <>
+        {dataSource &&
+          dataSource.map(record => {
+            return (
+              <Row style={{ minHeight: 50, textAlign: "left" }}>
```
Comment (nmhung1210) [inline src/shared/action/action-detail.tsx]: Add `key` prop for each element.

File: src/shared/action/action-detail.tsx [hunk 3]
```diff
@@ -145,13 +151,57 @@ class ActionDetailsInner extends PureComponent<Props> {
...
+    this.setState({ action, dataSource, actHash });
     return {
       action,
       dataSource
     };
   }
 
+  public renderActionData(): JSX.Element {
+    const { dataSource } = this.state;
+    return (
```
Comment (nmhung1210) [inline src/shared/action/action-detail.tsx]: check if `!dataSource` then return `null` here. An early return will gain more optimization.

File: src/shared/action/action-detail.tsx [hunk 4]
```diff
@@ -145,13 +151,57 @@ class ActionDetailsInner extends PureComponent<Props> {
...
+              </Row>
+            );
+          })}
+      </>
+    );
+  }
+
+  public renderCollpasable(): JSX.Element {
+    const { isTableFold, dataSource } = this.state;
+    return dataSource ? (
```
Comment (nmhung1210) [inline src/shared/action/action-detail.tsx]: Using `if (!dataSource) { return null; }` to make code more clean to read.

File: src/shared/action/action-detail.tsx (hunk 1 above)
Comment (puncsky) [reply src/shared/action/action-detail.tsx]: +1

File: src/shared/action/action-detail.tsx (hunk 4 above)
Comment (puncsky) [reply src/shared/action/action-detail.tsx]: +1

**Comment by gyanpgupta** [comment]
@puncsky  @nmhung1210  please have a look now

**Pull request merged**